packages/fgtclb/typo3-category-types/Classes/Loader/CategoryTypeLoader.php

```diff
@@ -58,8 +58,51 @@ public function loadUncached(): array
                 }
                 if (array_key_exists('types', $configArray) && is_array($configArray['types'])) {
                     foreach ($configArray['types'] as $categoryType) {
+                        // Check if the identifier and group of the category type to ensure it is unambiguous
+                        $categoryTypeIdentifier = $categoryType['identifier'] ?? null;
+                        if ($categoryTypeIdentifier === null) {
+                            throw new \Exception(
+                                'Category type identifier has to be defined.',
+                                1678979375330
+                            );
+                        }
+                        $categoryTypeGroup = $categoryType['group'] ?? null;
+                        if ($categoryTypeGroup === null) {
+                            throw new \Exception(
+                                'Category type group has to be defined.',
+                                1678979375330
+                            );
+                        }
+
+                        // Generate an unambiguous array key for the category type
+                        $categoryKey = $categoryTypeGroup . '.' . $categoryTypeIdentifier;
+
+                        // Remove a (default) category type if not needed in a project
+                        if (isset($categoryType['remove'])) {
+                            unset($loadedCategoryTypes[$categoryKey]);
+                            continue;
+                        }
+
+                        // The extension is added for debugging purposes
```

Comment:
```suggestion
                        // The extensionKey is added for debugging purposes
```

Temporarily ? Or should this maybe be hidden behind some kind of "context" switch ?

Reply:
Resolved, leaving open to be regonized.